### PR TITLE
[DOCS] 1.0 guide for getting sample data for testing or data exploration

### PR DESCRIPTION
**Pull request by Rachel-Reverie**
## Description
- Provides guidance for getting a Batch of data
   - from a Batch Definition
   - from the `pandas_default` Data Source's `.read_*()` methods.

Note: This code is not under test; it will be put under test in a follow up PR.

## Definition of done:
- [X] PR title is prefixed with one of: [BUGFIX], [FEATURE], [DOCS], [MAINTENANCE], [CONTRIB]
- [X] Code is linted - run `invoke lint` (uses `ruff format` + `ruff check`)

### CODE REVIEW

File: docs/docusaurus/docs/core/define_expectations/_retrieve_a_batch_of_test_data/_from_a_batch_definition.md [hunk 1]
```diff
@@ -0,0 +1,100 @@
...
+
+   ```python title="Python" name="docs/docusaurus/docs/core/define_expectations/_retrieve_a_batch_of_test_data/_from_a_batch_definition.py - retrieve Batch Definition"
+   ```
+
+2. Optional. Specify the Batch to retrieve.
+
+   Some Batch Definitions can only provide a single Batch.  Whole table batch definitions on SQL Data Assets, file path and whole directory Batch Definitions on filesystem Data Assets, and all Batch Definitions for dataframe Data Assets will provide all of the Data Asset's records as a single Batch.  For these Batch Definitions there is no need to specify which Batch to retrieve because there is only one available.
+
+   Yearly, monthly, and daily Batch Definitions subdivide the Data Asset's records by date.  This allows you to retrieve the data corresponding to a specific date from the Data Asset.  If you do not specify a Batch to retireve, these Batch Definitions will return the first valid Batch they find.  By default, this will be the most recent Batch (sort ascending) or the oldest Batch if the Batch Definition has been configured to sort descending.
```
Comment (tyler-hoffman) [inline docs/docusaurus/docs/core/define_expectations/_retrieve_a_batch_of_test_data/_from_a_batch_definition.md]: `retireve` 😄

**Review by tyler-hoffman — APPROVED** [review]
This all looks right to me!

File: docs/docusaurus/docs/core/define_expectations/_retrieve_a_batch_of_test_data/_from_a_batch_definition.md (hunk 1 above)
Comment (Rachel-Reverie) [reply docs/docusaurus/docs/core/define_expectations/_retrieve_a_batch_of_test_data/_from_a_batch_definition.md]: Good grief.  Thanks.  I really wish pycharm had better spellchecking (and grammarly support).  😅

Comment (tyler-hoffman) [reply docs/docusaurus/docs/core/define_expectations/_retrieve_a_batch_of_test_data/_from_a_batch_definition.md]: I have misspelled `checkpoint` as `checkpiont` in almost every PR I've pushed 😄

**Pull request merged**